torch/nn/functional.py

```diff
@@ -1308,6 +1318,53 @@ def layer_norm(input, normalized_shape, running_mean, running_var,
         return out
 
 
+def group_norm(input, num_groups, weight=None, bias=None, eps=1e-5):
+    r"""Applies Group Normalization for last certain number of dimensions.
+
+    See :class:`~torch.nn.GroupNorm` for details.
+    """
+
+    input_shape = input.size()
+    b = input_shape[0]
+    c = input_shape[1]
+    g = num_groups
+
+    if c % num_groups != 0:
+        raise ValueError('Expected number of channels in input to be divisible '
+                         'by num_groups, but got {} input and num_groups={}'
+                         .format(input_shape, num_groups))
+
+    if weight is not None and (weight.dim() != 1 or weight.numel() != c):
+        raise ValueError('Expected weight to be a vector of size equal to the '
+                         'number of channels in input, but got {} weight and {} '
+                         'input'.format(weight.size(), input_shape))
+
+    if bias is not None and (bias.dim() != 1 or bias.numel() != c):
+        raise ValueError('Expected bias to be a vector of size equal to the '
+                         'number of channels in input, but got {} bias and {} '
+                         'input'.format(bias.size(), input_shape))
+
+    # Apply group norm
+    input_reshaped = input.contiguous().view(1, b * g, -1)
```

Comment:
nit: this can be simplified with `.reshape(1, b * g, -1)`, which was added in https://github.com/pytorch/pytorch/pull/5575 and consists roughly as `.contiguous().view()`